sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1930,15 +1940,18 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
    */
   override def visitIntervalField(ctx: IntervalFieldContext): CalendarInterval = withOrigin(ctx) {
     import ctx._
-    val s = value.getText
+    val s = if (value.STRING() != null) {
+      string(value.STRING())
```

Comment:
This is to strip the `'` in the string, so that we don't need to deal with in the regex from `CalendarInterval`.